src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -806,26 +1128,35 @@ public void Reset()
 
             public ModelStateNode GetNode(StringSegment subKey, bool createIfNotExists)
             {
-                if (subKey.Length == 0)
-                {
-                    return this;
-                }
-
-                var index = BinarySearch(subKey);
                 ModelStateNode modelStateNode = null;
-                if (index >= 0)
-                {
-                    modelStateNode = ChildNodes[index];
-                }
-                else if (createIfNotExists)
+                if (subKey.Length != 0)
```

Comment:
What's the strategy here? Do you find that `ChildNodes ==` null is the common case and we can avoid the binary search when that happens?